### DISCUSSION TITLE
Conditional fields should be able to access values from other fields

### DISCUSSION THREAD

**Discussion by outline4** [craftcms/cms#11188] [Ideas] [answered]
**I would like to show certain fields only if I select a relation that has a certain value within that relation.** 

In the example below I only want to show the "Art" field (which is an entries field) if the category field (Kategorie) has the lightswitch "Art (switch)" selected. So I am missing a dropdown with something like "Field contains value x" where I can select a field from within that category field. 

I hope this makes sense and would be really powerful in my opinion. 

### Condition for the "Art" Field: 

In this screenshot there should be a dropdown with "contains value" and then another field where I can select the "Art (Switch)": 

<img width="1192" alt="Bildschirmfoto 2022-05-10 um 19 45 22" src="https://user-images.githubusercontent.com/145828/167710499-881fba43-0682-41c0-b7b3-078272c7b983.png">

Here you see the categories with the different switches on / off.  
So each category would only show the relevant fields: 

<img width="844" alt="Bildschirmfoto 2022-05-10 um 19 54 28" src="https://user-images.githubusercontent.com/145828/167711236-3e82e4bd-8fd3-4657-ae22-d6f717cd982d.png">

E.g.  if you select "Trauringe" in Kategorie the "Art" field would disappear and if you select "nach Kundenwunsch" the "Price" field would disappear because in that context they don't make any sense. 

This could make some serious fine grained backend experiences!  
I would love to see this come to life!

**Comment by brandonkelly** [comment]
If you expand the operator dropdown, there should be an “is related to” option, which allows you to enter a related element ID, or the name of an environment variable that is set to an ID.

An environment variable is recommended, because your element IDs are likely to be different depending on the environment.

**Reply by thupsi** [reply]
I think @outline4 was referring to something else: He would like to have a condition that is matched by **a field of** a related entry/category, not by the entry/category itself. At least this is how I understand it.

**Reply by brandonkelly** [reply]
Sorry, yeah, misunderstood.

**Comment by thupsi** [comment, marked as answer]
@outline4 I think a workaround for this should be possible using the [preparse field](https://github.com/besteadfast/craft-preparse-field) when it becomes available for Craft 4 (very soon, according to its author), and _if it is available for use in conditions_.

**Reply by brandonkelly** [reply]
Yep that could definitely work! The Preparse field could output `1` if the related category has the desired Lightswitch value, and then the Entries field could be conditional on whether the Preparse field has a value.

**Reply by outline4** [reply]
I will report back as soon as Preparse will be available.  

I still think it should be possible to select a field from within a relation.   
What are your thoughts on this? 

- [ ] fringe case 
- [x] frequent use case?

In my case it was the first thing I was trying to do... ha ha...

**Reply by brandonkelly** [reply]
I doubt it’s the last time someone will want to toggle a field based on whether another relation field is related to another element that meets some nested condition. Would be a tricky UI to get right, though, so can’t promise it’s something we’ll tackle.

**Reply by thupsi** [reply]
You could add a condition group/type, let's say ```nested conditions``` and then follow the existing pattern of chained dropdowns, with the first dropdown displaying only relationship fields. Something like that: ```nested condition: entry field -> related entry field -> ...```

It all sounds so simple when I' don't have to make it myself 😁

**Reply by outline4** [reply]
> Would be a tricky UI to get right, though, so can’t promise it’s something we’ll tackle.

UI wise I think it would be the easiest to just add a second line where you can chose the field you're interested in: 

<img width="733" alt="Bildschirmfoto 2022-05-13 um 13 41 27" src="https://user-images.githubusercontent.com/145828/168297305-65d1ed0a-d201-4aa6-b316-18810442ce25.png">

So you would choose "contains value" and then you'd be presented by the same possibilities as always. Preferably only show values that are within the relation field concerned. 

But what do I know :see_no_evil:

**Reply by thupsi** [reply]
Yes, I like this better. This is simpler still and more easily adaptable to other contexts (e.g custom sources). But ```contains value``` maybe is not appropriate, I would say something like ```nested condition```, or something along those lines.

**Reply by outline4** [reply]
Yes, I agree that the wording `nested condition` would be more appropriate.
